깃헙 CI 워크플로우 추가시 문제 처리

./github/workflow/main.yaml에서 
`run: yarn install`이나 `npm: npm install`에서 모두 이슈가 발생함으로 추가 조사 진행 필요. [여기 PR](https://github.com/bosagora/boa-space-seaport-js/pull/21) 참조

1. `run: yarn install`의 경우 다음의 에러 발생
Run yarn install
/__w/_temp/0bb985ed-2[4](https://github.com/bosagora/boa-space-seaport-js/actions/runs/4141227401/jobs/7160647059#step:6:5)27-4b2d-b8f8-4ccddaddc19e.sh: 1: yarn: not found
Error: Process completed with exit code 127.
로컬에서는 안 생기는 문제이고, 다시 조사할 필요 있음.

2. `run: npm install`의 경우 다음의 에러 발생
npm WARN tarball tarball data for ltgt@https://registry.npmjs.org/ltgt/-/ltgt-2.2.1.tgz (sha1-81ypHEk/e3PaDgdJUwTxezH4fuU=sha512-AI2r85+4MquTw9ZYqabu4nMwy9Oftlfa/e/52t9IjtfG+mGBbTNdAoZ3RQKLHR6r0wQnwZnPIEh/Ya6XTWAKNA==) seems to be corrupted. Trying again.
이 경우, 서버의 에러인것으로 보이고 기다려봐야할 것으로 보임.

--- a/.github/workflows/main.yaml
+++ b/.github/workflows/main.yaml
@@ -1,28 +1,32 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches: [ "v0.x.x" ]
+  pull_request:
+    branches: [ "v0.x.x" ]
 
 jobs:
   tests:
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        node-version: [16.x]
+
     steps:
       - uses: actions/checkout@v3
-      - name: Setup Node.js
-        uses: actions/setup-node@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
         with:
-          node-version: 16.11.0
+          node-version: ${{ matrix.node-version }}
 
-      # TODO: We should solve the problem as follows later
-      # error https://registry.yarnpkg.com/har-schema/-/har-schema-2.0.0.tgz: Extracting tar content of undefined failed, the file appears to be corrupt: "ENOENT: no such file or directory, open '/home/runner/.cache/yarn/v6/npm-har-schema-2.0.0-a94c2224ebcac04782a0d9035521f24735b7ec92-integrity/node_modules/har-schema/lib/header.json'"
+      - name: Install Dependencies
+        run: yarn install
 
-      # - name: Run install
-      #   uses: borales/actions-yarn@v4
-      #   with:
-      #     cmd: install # will run `yarn install` command
+      - name: Build
+        run: npm run build
 
-      # - name: Build
-      #   run: npm run build
-
-      # - name: Run tests
-      #   run: npm run test
+      - name: Run tests
+        run: npm run test

--- a/package.json
+++ b/package.json
@@ -58,7 +58,7 @@
   "dependencies": {
     "@0xsequence/multicall": "^0.39.0",
     "ethers": "^5.6.7",
-    "merkletreejs": "^0.2.31"
+    "merkletreejs": "^0.3.9"
   },
   "devDependencies": {
     "@nomiclabs/hardhat-ethers": "^2.0.5",
@@ -76,7 +76,7 @@
     "chai": "^4.3.6",
     "chai-as-promised": "^7.1.1",
     "concurrently": "^7.1.0",
-    "domain-registry": "git+https://github.com/ProjectOpenSea/domain-registry.git#70a5cb3dc919a13660dd83518fb2ab070009cce6",
+    "domain-registry": "ProjectOpenSea/domain-registry.git#70a5cb3dc919a13660dd83518fb2ab070009cce6",
     "dotenv": "^10.0.0",
     "eslint": "^8.17.0",
     "eslint-config-prettier": "^8.5.0",
@@ -98,7 +98,7 @@
     "sinon-chai": "^3.7.0",
     "solhint": "^3.3.7",
     "solidity-coverage": "^0.7.20",
-    "ts-node": "^10.9.1",
+    "ts-node": "^10.7.0",
     "typechain": "^7.0.1",
     "typescript": "^4.6.2"
   },

